docs/api-reference/index.rst

```diff
@@ -66,3 +66,25 @@ write but in order to avoid duplicated effort, they are defined by HPy.
    :maxdepth: 2
 
    inline-helpers
+
+Adding New API
+==============
+
+1. Add the function to ``hpy/tools/autogen/public_api.h`` and add the mapping
+   from the function to the cpython equivalent to ``hpy/tools/autogen/conf.py``.
+
+2. If the function cannot be autogenerated (i.e. the mapping does not exist),
+   you must write the wrapper by hand. Add the function to ``NO_WRAPPER`` in
+   ``hpy/tools/autogen/debug.py``, and add a ``ctx_fname`` function to
+   ``hyp/devel/src/runtime/*.c`` (possibly adding the new file to ``setup.py``),
+   add a debug wrapper to ``hpy/debug/src/debug_ctx.c``, add a implementation
+   that uses the `ctx` variant to ``hpy/devel/include/hpy/cpython/misc.h`` and
+   add the declaration to ``hpy/devel/include/hpy/runtime/ctx_funcs.h``.
+
+3. Run ``make autogen`` which will turn the mapping into autogenerated functions
+
+4. Add a test for the functionality
+
+5. Build with ``python setup.py build_ext``. After that works, build with
```

Comment:
You should use `make` instead of `python setup.py build_ext`. If you really want to directly run `setup.py`, then it currently is:
```
python3 setup.py build_clib -f build_ext -if
```
Reason: `build_clib` will also build the static lib for the context that will speed up the tests a lot. But yes, your comamnd still works.

Also, installation is not required to run tests if you use `make` (which builds in-place).

Reply:
I don't think `make` works on windows. I use the minimal `build_ext` on purpose, since it is the quickest way to flush out build errors, and once that works recommend using the command used in the CI jobs.

Reply:
I agree, `make` won't probably work on Windows (I think it could but that's rather tricky, irrc). Anyway, using `build_ext`to get compilation errors is fine. You can also run all tests, it's just slower.